unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Catalog/SceneObjectCatalogController.cs

```diff
@@ -0,0 +1,389 @@
+    List<SceneObject> favoritesSceneObjects = new List<SceneObject>();
+    List<SceneObject> quickBarShortcutsSceneObjects = new List<SceneObject>() { null, null, null,null,null,null,null,null,null };
+    private void Start()
+    {
+        OnResultReceived += AddFullSceneObjectCatalog;
+        catalogAssetPackListView.OnSceneAssetPackClick += OnScenePackSelected;
+        catalogGroupListView.OnSceneObjectClicked += SceneObjectSelected;
+        catalogGroupListView.OnSceneObjectFavorite += AsignFavorite;
+        catalogGroupListView.OnAdapterStartDragging += SceneObjectStartDragged;
+        catalogGroupListView.OnAdapterDrag += OnDrag;
+        catalogGroupListView.OnAdapterEndDrag += OnEndDrag;
+        searchInputField.onValueChanged.AddListener(OnSearchInputChanged);
+    }
+
+    private void OnDestroy()
+    {
+        catalogAssetPackListView.OnSceneAssetPackClick -= OnScenePackSelected;
+        catalogGroupListView.OnSceneObjectClicked -= SceneObjectSelected;
+    }
+
+    #region Filter
+    void OnSearchInputChanged(string currentSearchInput)
+    {
+        if (string.IsNullOrEmpty(currentSearchInput))
+        {
+            ShowAssetsPacks();
+        }
+        else
+        {
+            ShowCatalogContent();
+            FilterAssets(currentSearchInput);
+            catalogGroupListView.SetContent(filterObjects);
+        }
+        lastFilterName = currentSearchInput;
+    }
+
+    void FilterAssets(string nameToFilter)
+    {
+        filterObjects.Clear();
+        foreach (SceneAssetPack assetPack in CatalogController.sceneObjectCatalog.GetValues().ToList())
+        {
+            foreach (SceneObject sceneObject in assetPack.assets)
+            {
+                if (sceneObject.category.Contains(nameToFilter) || sceneObject.tags.Contains(nameToFilter) || sceneObject.name.Contains(nameToFilter))
+                {
+                    bool foundCategory = false;
+                    foreach (Dictionary<string, List<SceneObject>> groupedSceneObjects in filterObjects)
+                    {
+                        if (groupedSceneObjects.ContainsKey(sceneObject.category))
+                        {
+                            foundCategory = true;
+                            if (!groupedSceneObjects[sceneObject.category].Contains(sceneObject))
+                                groupedSceneObjects[sceneObject.category].Add(sceneObject);
+                        }
+                    }
+                    if (!foundCategory)
+                    {
+                        Dictionary<string, List<SceneObject>> groupedSceneObjects = new Dictionary<string, List<SceneObject>>();
+                        groupedSceneObjects.Add(sceneObject.category, new List<SceneObject>() { sceneObject });
+                        filterObjects.Add(groupedSceneObjects);
+                    }
+                }
+            }
+        }
+    }
```

Comment:
As you wish! Take into account that this function is called each time that a user types a new character in the search bar

Reply:
Uh... Good that you mention it because I didnt think about that. There's probably room for improvement in performance by carrying the filter phrase and analyze when a character is added/removed but that would make it overwhelmingly complex.

Let's leave it as it is and just give it a shot at creating some sub-methods to increase readability.